packages/@react-stately/data/docs/useAsyncList.mdx

```diff
@@ -174,6 +174,54 @@ let list = useAsyncList({
 });
 ```
 
+## Filtering
+
+There are many instances where your list of data may need to be filtered, such as during user lookup or query searches.
+For server side filtering, this can be implemented by leveraging the `filterText` passed to the `load` as an API parameter.
+`useAsyncList` returns the state variable `filterText` and its setter `setFilterText`, allowing you to update the
+`filterText` used within `load` in response to any updates within your component.
+
+### Server side filtering
+
+The example below shows how server side filtering could be implemented by using `filterText` in the `load` function and passing a parameter to the API.
+The input text of the ComboBox is controlled by providing `list.filterText` as the ComboBox's `inputValue` prop. By passing `list.setFilterText` to
+the ComboBox via the `onInputChange` prop, the `filterText` within `load` is updated whenever the end user modifies the ComboBox's
+text, ensuring that the list of items is properly updated. Finally, the state of the list (`loading`, `loadingMore`, `filtering`, etc) is
+communicated to the ComboBox through `list.state`, allowing the ComboBox to display the appropriate visual feedback to an end user.
+
+```tsx
+let list = useAsyncList({
+  async load({signal, cursor, filterText}) {
+    if (cursor) {
+      cursor = cursor.replace(/^http:\/\//i, 'https://');
+    }
+
+    // If no cursor is available, then we're loading the first page,
+    // filtering the results returned via a query string that
+    // mirrors the ComboBox input text.
+    // Otherwise, the cursor is the next URL to load,
+    // as returned from the previous page.
+    let res = await fetch(cursor || `https://swapi.dev/api/people/?search=${filterText}`, {signal});
```

Comment:
I think we could remove the cursor stuff and loadMore stuff from this example to simplify it and keep it focused on filtering specifically.